### PR TITLE
fix: impl `Drop` for `RedisConnectionPool`

### LINKED ISSUE #1035: [BUG] panic on RedisPoolConnection close.
### Bug Description

Currently the RedisPoolConnection close panics on [this line](https://github.com/juspay/hyperswitch/blob/ed99655ebc11d53f4b2ffcb8c0eb9ef6b56f32c4/crates/router/src/bin/router.rs#L47). The root cause of this issue is `Arc::get_mut` throws None when `Arc` has 2 or more strong immutable reference. Since we are creating async tasks with reference to `Arc<RedisConnectionPool>` [here](https://github.com/juspay/hyperswitch/blob/ed99655ebc11d53f4b2ffcb8c0eb9ef6b56f32c4/crates/router/src/services.rs#L110).

### Expected Behavior

Runs successfully and closes the connection to redis and db connection.

### Actual Behavior

```text
thread 'main' panicked at 'Redis connection pool cannot be closed', crates/router/src/db.rs:74:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

### Steps To Reproduce

1. Run `cargo r`
2.  Stop it with `CTRL+C` 
3. You will get:

   ```text
   thread 'main' panicked at 'Redis connection pool cannot be closed', crates/router/src/db.rs:74:14
   note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
   ```

### Possible Resolution

Possible resolution would be removing `&mut self` from close function. Since `redis.quit_pool` doesn't really need mutable reference. Other one would be something like.
```rust
impl Drop for RedisConnectionPool {
	fn drop(self) {
		tokio_rt.block(async {
			// call to the function that closes the connections
		})
	}
}
```

### Environment

Are you using hyperswitch hosted version? No

If not (or if building/running locally), please provide the following details:
1. Operating System or Linux distribution: MacOS
2. Rust version (output of `rustc --version`): ``1.69
3. App version (output of `cargo r -- --version`): ``0.5.9

### Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

None

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Bugfix

## Description

This fixes #1035 

## Motivation and Context

#1035 

## How did you test it?

Manual

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

**Review by jarnura — DISMISSED** [review]

File: crates/router/src/bin/router.rs [hunk 1]
```diff
@@ -39,13 +39,11 @@ async fn main() -> ApplicationResult<()> {
     logger::info!("Application started [{:?}] [{:?}]", conf.server, conf.log);
 
     #[allow(clippy::expect_used)]
-    let (server, mut state) = router::start_server(conf)
+    let (server, _) = router::start_server(conf)
```
Comment (NishantJoshi00) [inline crates/router/src/bin/router.rs:42]: If we don't need the state here, then can't we change the function signature of `start_server`

**Review by NishantJoshi00 — APPROVED** [review]
Left an optional comment, not an immediate priority. Everything looks good!

File: crates/router/src/bin/router.rs (hunk 1 above)
Comment (Abhicodes-crypto) [inline crates/router/src/bin/router.rs:42]: Sorry for interrupting but I have a small question, 

In `start_server` function the returned value is a clone of the `state`. 

So here if a `state` clone is dropped, and that triggers the redis_connection pool drop function. 

Does that mean anywhere in the code base if a `state` clone is dropped then redis connection is dropped(closed) ?

Comment (dracarys18) [reply crates/router/src/bin/router.rs:42]: Since state has `Arc<RedisConnectionPool>` dropping would decrement the reference counter

Comment (Abhicodes-crypto) [reply crates/router/src/bin/router.rs]: Got it. Thanks.

**Review by Abhicodes-crypto — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**